### PR TITLE
PullThroughReduce and ReduceReshapeFusion transformations - handle more than one consumers

### PR DESCRIPTION
**Pull request by mbencer**
Ticket: 96832

### CODE REVIEW

**Review by mitruska — APPROVED** [review]

**Review by artkuli — APPROVED** [review]
LGTM!

File: src/common/transformations/src/transformations/common_optimizations/pull_through_reduce.cpp [hunk 1]
```diff
@@ -103,10 +103,11 @@ ov::pass::PullUnsqueezeThroughReduce::PullUnsqueezeThroughReduce() {
 
     const auto input = pattern::any_input(pattern::has_static_rank());
     const auto unsqueeze_axes = pattern::wrap_type<opset9::Constant>();
-    const auto unsqueeze = pattern::wrap_type<opset9::Unsqueeze>({input, unsqueeze_axes}, pattern::has_static_rank());
+    const auto unsqueeze = pattern::wrap_type<opset9::Unsqueeze>({input, unsqueeze_axes}, pattern::consumers_count(1));
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/pull_through_reduce.cpp:106]: Do we have "pattern::add" or something like that?

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/pull_through_reduce.cpp [hunk 2]
```diff
@@ -103,10 +103,11 @@ ov::pass::PullUnsqueezeThroughReduce::PullUnsqueezeThroughReduce() {
...
     const auto reduce_axes = pattern::wrap_type<opset9::Constant>();
     const auto reduce = pattern::wrap_type<op::util::ArithmeticReductionKeepDims, op::util::LogicalReductionKeepDims>(
-        {unsqueeze, reduce_axes});
+        {unsqueeze, reduce_axes},
+        pattern::consumers_count(1));
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/pull_through_reduce.cpp:110]: Is `consumers_count(1)` really a limitation here?
as I can see we use `replace_node(m.get_match_root(), new_unsqueeze_node);` , it should handle the case when there are multiple consumers of  Reduction op

Comment (mbencer) [reply src/common/transformations/src/transformations/common_optimizations/pull_through_reduce.cpp]: @itikhono A good catch, thank you. Restrictions removed.

**Pull request merged**